astroplan/utils.py

```diff
@@ -1,10 +1,118 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
+import warnings
+import numpy as np
+from astropy.utils.data import download_file, clear_download_cache
+from astropy.utils import iers
+from astropy.time import Time
+import astropy.units as u
+from .exceptions import OldEarthOrientationDataWarning
+from astropy.utils.data import (_get_download_cache_locs, CacheMissingWarning,
+                                _open_shelve)
 
-__all__ = ["_mock_remote_data"]
+__all__ = ["get_IERS_A_or_workaround", "download_IERS_A"]
+
+IERS_A_WARNING = ("For best precision (on the order of arcseconds), you must "
+                  "download an up-to-date IERS Bulletin A table. To do so, run:"
+                  "\n\n"
+                  ">>> from astroplan import download_IERS_A\n"
+                  ">>> download_IERS_A()\n")
+
+BACKUP_Time_get_delta_ut1_utc = Time._get_delta_ut1_utc
 
 def _mock_remote_data():
     # Mock FixedTarget.from_name class method for tests without remote data
     from astroplan.core import FixedTarget
     FixedTarget.from_name = FixedTarget._fixed_target_from_name_mock
+
+def _low_precision_utc_to_ut1(self, jd1, jd2):
+    """
+    When no IERS Bulletin A is available (no internet connection), use low
+    precision time conversion by assuming UT1-UTC=0 always.
+    This method mimics `~astropy.coordinates.builtin_frames.utils.get_dut1utc`
+    """
+    try:
+        return self.delta_ut1_utc
+    except IndexError:
+        warnings.warn(IERS_A_WARNING, OldEarthOrientationDataWarning)
+        return np.zeros(self.shape)
+
+def get_IERS_A_or_workaround():
+    """
+    Get the cached IERS Bulletin A table if one exists. If one does not exist,
+    monkey patch `~astropy.time.Time._get_delta_ut1_utc` so that
+    `~astropy.time.Time` objects don't raise errors by computing UT1-UTC off
+    the end of the IERS table.
+    """
+    if IERS_A_in_cache():
+        iers.IERS.iers_table = _get_IERS_A_table()
+    else:
+        Time._get_delta_ut1_utc = _low_precision_utc_to_ut1
+
+def IERS_A_in_cache():
+    """
+    Check if the IERS Bulletin A table is locally cached.
+    """
+    url_key = iers.IERS_A_URL
+    # The below code which accesses ``urlmapfn`` is stolen from
+    # astropy.utils.data.download_file()
+    try:
+        dldir, urlmapfn = _get_download_cache_locs()
+    except (IOError, OSError) as e:
+        msg = 'Remote data cache could not be accessed due to '
+        estr = '' if len(e.args) < 1 else (': ' + str(e))
+        warnings.warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
+        return False
+    with _open_shelve(urlmapfn, True) as url2hash:
+        if url_key in url2hash:
```

Comment:
yes, @bmorris3, that's what seemed to work for me.  There's the possibility that this will get us into trouble in ways I don't really understand (a lot of py 2.x unicode stuff is basically just "try it till it works" in my mind...), but this at least will keep shelve from freaking out.

As for a test, I'm not sure how to really test it robustly on different platforms...

If it helps, though, I guess the "test" gets run anytime someone does any tests because the import of astroplan will invoke this, right?